scripts/utils.py

```diff
@@ -82,23 +77,83 @@ def _getFileNameDisplayString(self):
         else:
             return self.basename  # TODO scrolling window here
 
-    def _updateDisplay(self):
+    def _updateDisplay(self, modulo=1):
+        self.displayCounter += 1
+        if self.displayCounter % modulo != 0:
+            return
         fileName = self._getFileNameDisplayString()
-
-        # TODO contentLength seems to slightly under-report how many bytes
+        # TODO contentlength seems to slightly under-report how many bytes
         # we have to download... hence the min functions
-        percentage = min(self.bytesWritten / self.contentLength, 1)
+        percentage = min(self.bytesReceived / self.fileSize, 1)
         numerator = humanize.filesize.naturalsize(
-            min(self.bytesWritten, self.contentLength))
+            min(self.bytesReceived, self.fileSize))
         denominator = humanize.filesize.naturalsize(
-            self.contentLength)
-
+            self.fileSize)
         displayString = "{}   {:<6.2%} ({:>9} / {:<9})\r"
         self.stream.write(displayString.format(
             fileName, percentage, numerator, denominator))
         self.stream.flush()
 
 
+class HttpFileDownloader(FileDownloader):
+    """
+    Provides a wget-like file download and terminal display for HTTP
+    """
+    defaultChunkSize = 1048576  # 1MB
+
+    def __init__(self, url, path, chunkSize=defaultChunkSize,
+                 stream=FileDownloader.defaultStream):
+        super(HttpFileDownloader, self).__init__(
+            url, path, stream)
+        self.chunkSize = chunkSize
+
+    def download(self):
+        self._printStartDownloadMessage()
+        response = requests.get(self.url, stream=True)
+        response.raise_for_status()
+        self.fileSize = int(response.headers['content-length'])
+        with open(self.path, 'wb') as outputFile:
+            for chunk in response.iter_content(chunk_size=self.chunkSize):
+                self.bytesReceived += self.chunkSize
+                self._updateDisplay()
+                outputFile.write(chunk)
+        self._cleanUp()
+
+
+class FtpFileDownloader(FileDownloader):
```

Comment:
Are we using this now? If not we should remove it. It might be worth considering dropping the changes to this file entirely if we're not actually using them now .